src/util/xml_expr.cpp

```diff
@@ -273,7 +274,7 @@ xmlt xml(
     {
       const struct_typet &struct_type=to_struct_type(type);
       const struct_typet::componentst &components=struct_type.components();
-      assert(components.size()==expr.operands().size());
+      CHECK_RETURN(components.size()==expr.operands().size());
```

Comment:
`DATA_INVARIANT(..., "A struct expression should have an operand for each member of its struct type")`?